modules/default/weather/providers/smhi.js

```diff
@@ -17,41 +17,56 @@ WeatherProvider.register("smhi", {
 	defaults: {
 		lat: 0,
 		lon: 0,
-		precipitationValue: "pmedian"
+		precipitationValue: "pmedian",
+		location: false,
 	},
 
 	/**
-	 * Implements method in interface for fetching current weather
+	 * Implements method in interface for fetching current weather.
 	 */
 	fetchCurrentWeather() {
 		this.fetchData(this.getURL())
 			.then((data) => {
 				let closest = this.getClosestToCurrentTime(data.timeSeries);
 				let coordinates = this.resolveCoordinates(data);
 				let weatherObject = this.convertWeatherDataToObject(closest, coordinates);
-				this.setFetchedLocation(`(${coordinates.lat},${coordinates.lon})`);
+				this.setFetchedLocation(this.config.location || `(${coordinates.lat},${coordinates.lon})`);
 				this.setCurrentWeather(weatherObject);
 			})
 			.catch((error) => Log.error("Could not load data: " + error.message))
 			.finally(() => this.updateAvailable());
 	},
 
 	/**
-	 * Implements method in interface for fetching a forecast.
-	 * Handling hourly forecast would be easy as not grouping by day but it seems really specific for one weather provider for now.
+	 * Implements method in interface for fetching a multi-day forecast.
 	 */
 	fetchWeatherForecast() {
 		this.fetchData(this.getURL())
 			.then((data) => {
 				let coordinates = this.resolveCoordinates(data);
-				let weatherObjects = this.convertWeatherDataGroupedByDay(data.timeSeries, coordinates);
-				this.setFetchedLocation(`(${coordinates.lat},${coordinates.lon})`);
+				let weatherObjects = this.convertWeatherDataGroupedBy(data.timeSeries, coordinates);
+				this.setFetchedLocation(this.config.location || `(${coordinates.lat},${coordinates.lon})`);
 				this.setWeatherForecast(weatherObjects);
 			})
 			.catch((error) => Log.error("Could not load data: " + error.message))
 			.finally(() => this.updateAvailable());
 	},
 
+	/**
+	 * Implements method in interface for fetching hourly forecasts.
+	 */
+	fetchWeatherHourly() {
+		this.fetchData(this.getURL())
+			.then((data) => {
+				let coordinates = this.resolveCoordinates(data);
```

Comment:
could be `const` I guess?

Reply:
Yes, I agree in both cases! I was torn between using `const` and sticking to the [original author's code style](https://github.com/MichMich/MagicMirror/blob/0a1067ec7d72271a6296ca4b8a0ad392e1ce90fa/modules/default/weather/providers/smhi.js#L47). Since there are no code style guidelines except for formatting, I thought it'd be best to stick to existing patterns.

That said, I'd more than happily change to `const`, at the very least when it comes to my own contributions!